docs/source/concepts/identity-and-package-management.rst

```diff
@@ -0,0 +1,169 @@
+Currently, the Ledger API provides no authentication mechanisms.
+However, it will soon support authentication through JSON Web Tokens.
+Before the newly provisioned party can be used, the application will have to obtain a token for this party.
+The issuance of tokens is specific to each ledger and independent of the Ledger API.
+The same is true for the policy which the participants use to decide whether to accept a token.
+
+.. _identifiers-and-real-world:
+
+Identifiers and the Real World
+==============================
+
+The "substrate" on which DAML workflows are built are the real-world obligations of the parties in the workflow.
+To give value to these obligations, they must be connected to parties in the real world.
+However, the process of linking party identifiers to real-world entities is left to the ledger implementation.
+
+A :ref:`global state topology <global-state-topologies>` might simplify the process by trusting the operator of the writer node(s) with providing the link to the real world.
+For example, if the operator is a stock exchange, it might guarantee that a real-world exchange participant whose legal name is "Bank Inc." is represented by a ledger party with the identifier "Bank Inc.".
+Alternatively, it might use a random identifier, but guarantee that the display name is "Bank Inc.".
+Ledgers with :ref:`partitioned topologies <partitioned-topologies>` in general might not have such a single store of identities.
+The solutions for linking the identifiers to real-world identities could rely on certificate chains, `verifiable credentials <https://www.w3.org/TR/vc-data-model/>`__, or other mechanisms.
+
+Package Management
+******************
+
+All DAML Ledgers implement endpoints that allow for provisioning new DAML code to the ledger.
+The vetting process for this code, however, depends on the particular ledger implementation and its configuration.
+The remainder of this section describes the endpoints and general principles behind the vetting process.
+The details of the process are ledger-dependent.
+
+Package Formats and Identifiers
+===============================
+
+Any code -- i.e., DAML templates -- to be uploaded must compiled down to the :ref:`DAML-LF <daml-lf>` language.
+The unit of packaging for DAML-LF is the :ref:`.dalf <dar-file-dalf-file>` file.
+Each ``.dalf`` file is uniquely identified by its **package identifier**, which is the hash of its contents.
+Templates in a ``.dalf`` file can references templates from other ``.dalf`` files, i.e., ``.dalf`` files can depend on other ``.dalf`` files.
+A :ref:`.dar <dar-file-dalf-file>` file is a simple archive containing multiple ``.dalf`` files, and has no identifier of its own.
+The archive provides a convenient way to package ``.dalf`` files together with their dependencies.
+The Ledger API supports only ``.dar`` file uploads.
```

Comment:
Add: However storage of data on-ledger typically happens using `.dalf` files, as only they have globally unique identifiers.

Reply:
Added